### PR TITLE
Use common skinnable drawable for catcher and catcher trails

### PR DESCRIPTION
**Pull request by ekrctb**
- Fix catcher trails are not updated when skin is changed

### CODE REVIEW

**Comment by peppy** [comment]
Hmm, one of the things which we eventually hoped to add was character turn animations, where you definitely wouldn't want the animation to be transferred to the trail (we already have the resources for this somewhere).

Not sure about this direction, since the previous implementation was intentionally freezing the frame at the point of trail generation.

If we're going to go with this, then I'd consider disabling animations completely because this isn't how it should work conceptually..

**Comment by bdach** [comment]
As for the original issue of trail sprites not updating on skin change, I'd probably suggest simply purging them on skin change for simplicity.

**Comment by ekrctb** [comment]
To be fair, the animation freezing is ignored only for temporary, as far as I'm anticipating. This approach is perfectly capable of implementing the animation freezing, by using a frozen clock for trails.

The thing is, the current way to managing the animation (reset animation frame when the state changes) is broken when a replay is rewound. Because I want to fix that, the way to do the animation is going to change soon.
And once a way to manage animation state that is correctly rewinding is implemented, the animation freezing should be easier to implement.

**Comment by peppy** [comment]
I haven't gone through the code in detail, but if it's a temporary thing definitely add an inline comment mentioning that this needs to be fixed at a later stage.

If the eventual goal is to bring back frame locking then I'll probably be more okay with this change.

**Comment by peppy** [comment]
On a quick search I came across at least one skin using animated sprites (https://osu.ppy.sh/community/forums/topics/839786) and I'm pretty sure there will be others. With this PR it looks *very* weird, to the point I'm hesitant to merge this unless it's fixed locally. I think it shouldn't be a huge amount of extra changes to fix it in this PR?

File: osu.Game.Rulesets.Catch/UI/SkinnableCatcher.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace osu.Game.Rulesets.Catch.UI
+{
+    /// <summary>
+    /// The visual representation of the <see cref="Catcher"/>.
+    /// It includes the body part of the catcher and the catcher plate.
+    /// </summary>
+    public class SkinnableCatcher : SkinnableDrawable
+    {
+        [Cached]
+        public readonly Bindable<CatcherAnimationState> AnimationState = new Bindable<CatcherAnimationState>();
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/SkinnableCatcher.cs:20]: May be worth adding xmldoc here mentioning that this is (cached) here as there may be several uses of this drawable in different animation states (ie. for the catcher trail).

**Comment by ekrctb** [comment]
I made the animation of the catcher trails frozen.
It is not currently using the right frame of the animation (always using the first frame), but it should look better.

**Review by peppy — APPROVED** [review]

**Pull request merged**